koku/masu/processor/tasks.py

```diff
@@ -421,9 +443,24 @@ def update_cost_model_costs(schema_name, provider_uuid, start_date=None, end_dat
         updater.update_cost_model_costs(start_date, end_date)
 
 
-@app.task(name="masu.processor.tasks.refresh_materialized_views", queue_name="reporting")
-def refresh_materialized_views(schema_name, provider_type, manifest_id=None):
+@app.task(
+    name="masu.processor.tasks.refresh_materialized_views",
+    queue_name="reporting",
+    autoretry_for=(TaskRunningError,),
+    retry_backoff=True,
+    retry_backoff_max=10,
+    max_retries=100,
+    bind=True,
+)
+def refresh_materialized_views(self, schema_name, provider_type, manifest_id=None):
     """Refresh the database's materialized views for reporting."""
+    task_id = None
+    if self.request:
+        task_id = self.request.id
+    if is_task_currently_running("masu.processor.tasks.refresh_materialized_views", task_id, [schema_name]):
+        msg = f"Already running refresh_materialized_views for {schema_name}."
+        raise TaskRunningError(msg)
```

Comment:
Is this an error that we need to raise an exception on? Or is this just a condition where we need to not execute the materialized view refresh code?

What purpose does raising an exception here serve?

Reply:
Comments added, and I wish there was a better way, but celery needs the task to fail with an exception to auto-retry.